scrapers/faculty_directory_scraper/main.py

```diff
@@ -3,43 +3,126 @@
 import time
 import requests
 import aiohttp
+from lxml import html
+from lxml import etree
 from bs4 import BeautifulSoup
 import json
+import re
 
 
-async def get_professor(s, professor, url, data):
-    async with s.get(url) as response:
+BASE_URL = "https://faculty.rpi.edu"
+
+
+def clean_string(string):
+    # Remove newlines and nbsp, as well as repeated spaces.
+    string = re.sub("[\n\u00a0 ]+", " ", string)
+    # "founder of lightexture , where she designs"
+    # =>
+    # "founder of lightexture, where she designs"
+    # https://faculty.rpi.edu/yael-erel
+    return re.sub(" +([,.])", r"\1", string).strip()
+
+
+async def get_professor(session, url, data):
+    async with session.get(f"{BASE_URL}/{url}") as response:
+        print(f"Scraping {url}...")
         soup = BeautifulSoup(await response.text("utf8"), "lxml")
-        data[professor["node"]["Path"]] = {}
-        professor_data = data[professor["node"]["Path"]]
-        professor_data["name"] = professor["node"]["title"].strip()
-
-        # Find all fields on the professor page and append them to the data object
-        for item in soup.findAll("div", {"class": "views-field"}):
-            class_parts = item["class"][1].split("-")
-            if class_parts[len(class_parts) - 1] == "portrait":
-                professor_data[class_parts[len(class_parts) - 1]] = item.find("img")[
-                    "src"
-                ].split("?")[0]
-            elif item.find("div", {"class": "field-content"}):
-                professor_data[class_parts[len(class_parts) - 1]] = " ".join(
-                    item.find("div", {"class": "field-content"}).get_text(" ").split()
+        entry = {}
+        entry["name"] = soup.find("span", {"class": "field--name-title"}).text
+        entry["portrait"] = BASE_URL + soup.find("img", {"class": "img-fluid"})["src"]
+
+        # Sometimes the title is in field--name-field-title, and sometimes
+        # it is in field--name-field-alternate-title. No explanation for that.
+        if not (title := soup.find("div", {"class", "field--name-field-title"})):
+            title = soup.find("div", {"class", "field--name-field-alternate-title"})
```

Comment:
Could this be simplified down to (legit question, I dont know)
```suggestion
        title = soup.find("div", {"class", "field--name-field-title"}) or soup.find("div", {"class", "field--name-field-alternate-title"})
```
This use of the `:=` operator mixed with the `if not` statement just makes this hard to read imo

Reply:
I think it can; I completely forgot about `or`. Good idea